tensorly/metrics/factors.py

```diff
@@ -28,16 +33,33 @@ def congruence_coefficient(matrix1, matrix2, absolute_value=True):
     congruence : float
     permutation : list
     """
-    num_cols = T.shape(matrix1)[1]
-    if T.shape(matrix1) != T.shape(matrix2):
-        raise ValueError("Matrices must have same shape")
-
-    matrix1 = matrix1 / T.norm(matrix1, axis=0)
-    matrix2 = matrix2 / T.norm(matrix2, axis=0)
-    all_congruences = T.dot(T.transpose(matrix1), matrix2)
-    if absolute_value:
-        all_congruences = T.abs(all_congruences)
-    all_congruences = T.to_numpy(all_congruences)
+    # Check if matrix1 and matrix2 are lists of the same length
+    if isinstance(matrix1, list):
+        if not isinstance(matrix2, list) or len(matrix1) != len(matrix2):
+            raise ValueError("Input lists of matrices must have the same length")
+    else:
+        matrix1 = [matrix1]
+        matrix2 = [matrix2]
+    all_congruences_list = []
+    # Store an arbitrary column dimension, they should all be the same
+    num_col_mat1 = T.shape(matrix1[0])[1]
+    for mat1, mat2 in zip(matrix1, matrix2):
+        num_cols = T.shape(mat1)[1]
+        if T.shape(mat1) != T.shape(mat2):
+            raise ValueError("Matrices must have same shape")
+        if T.shape(mat1)[1] != num_col_mat1 or T.shape(mat2)[1] != num_col_mat1:
+            # check that all matrices have the same number of columns
+            raise ValueError("Matrices must have the same number of columns")
+
+        mat1 = mat1 / T.norm(mat1, axis=0)
```

Comment:
Should we add an `eps` (we can use tl.eps) or check for zero norm?

Reply:
I guess we should not allow 0 norm in the columns of inputs, so I will add a check for that.

Reply:
I guess we could even just wrap those two divisions in a try/except and just catch `ZeroDivisionError` -- better ask for forgiveness than permission.

Reply:
Turns out e.g. with numpy backend we don't have a ZeroDivisionError but a RuntimeWarning; and mat1 and mat2 are filled with NaNs. I am not confident in my knowledge of warnings with the different backends, so I am not sure how to catch that properly :/

Reply:
I guess we can just explicitly check if T.norm(...) > 0